### ISSUE TITLE
build artifact names use PyPa version not semver

### ISSUE THREAD

**Issue by eyablonowitz** [fleetingbytes/hatch-semver#89]
When I build with a prerelease version I get an artifact named with PyPa versioning not semver.  Is this expected?

```
% hatch version 
0.1.3
% hatch version alpha
Old: 0.1.3
New: 0.1.4-alpha.1
% hatch build -t sdist
...
dist/dblunch-0.1.4a1.tar.gz
```

pyproject.toml:
```
[build-system]
requires = ["hatchling", "hatch-semver"]
build-backend = "hatchling.build"
...
[project]
name = "dblunch"
...
dynamic = ["version"]
...
[tool.hatch.version]
path = "src/dblunch/__about__.py"
validate-bump = true
scheme = "semver"
...
```

**Comment by fleetingbytes** [comment]
That is correct. PyPI accepts only Python's versioning scheme. Semantic versions are translated to it: https://packaging.python.org/en/latest/specifications/version-specifiers/#semantic-versioning

**Comment by eyablonowitz** [comment]
Thanks @fleetingbytes. That makes sense.  FWIW, in my use case I am distributing a private package via Homebrew. I am not using PyPi at all.

Homebrew infers the version from the artifact name and the PyPa versioning throws it off when comparing versions:
For example it thinks that `dblunch-0.1.4a6.tar.gz` is newer than `dblunch-0.1.5.tar.gz`.

Looks like as a workaround I can provide the semver explicitly to Homebrew:
https://rubydoc.brew.sh/Formula#version-class_method.

**Comment by fleetingbytes** [comment]
That's interesting. Maybe hatch could be told to maintain another versioning scheme in the build filenames. Let's see what @ofek thinks about this.

**Comment by ofek** [comment]
Feel free to open a feature request asking for an option similar to `strict-naming` https://hatch.pypa.io/latest/plugins/builder/wheel/#options (also for source distributions)

**Comment by eyablonowitz** [comment]
Thanks @ofek and @fleetingbytes.  Hatch feature request:
https://github.com/pypa/hatch/issues/1251
